File: cases/taint/scala/call-context-positive/CallContextPositive.scala
package dataflowbench

object CallContextPositive {
  def dfb_source(): Int = { // DFB-SOURCE: call-context-input
    1
  }

  def relay(value: Int): Int = { // DFB-WITNESS: call-context-relay
    value
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: call-context-sink

  def run(): Unit = {
    val tainted = relay(dfb_source())
    val clean = relay(0)
    dfb_sink(tainted)
  }
}

File: cases/taint/csharp/call-context-positive/CallContextPositive.cs
namespace DataFlowBench;

static class CallContextPositive
{
    static int dfb_source() // DFB-SOURCE: call-context-input
    {
        return 1;
    }

    static int Relay(int value) // DFB-WITNESS: call-context-relay
    {
        return value;
    }

    static void dfb_sink(int value) { } // DFB-SINK: call-context-sink

    static void Run()
    {
        int tainted = Relay(dfb_source());
        int clean = Relay(0);
        dfb_sink(tainted);
    }
}
